### PR TITLE
Docs: Add documentation for `next/third-parties`

### PR DESCRIPTION
**Pull request by housseindjirdeh**
This PR adds documentation for `next/third-parties`.

- [X] Add a page to the Optimizing section of the docs
- [ ] Add a page to the API reference section of the docs that shows all required and supported props for each third-party
- [ ] Test changes locally

### CODE REVIEW

**Review by Sanjay-shankar — APPROVED** [review]
Looks good to Merge

File: docs/02-app/01-building-your-application/06-optimizing/10-third-parties.mdx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+```
+
+## Google Third-Parties
+
+All supported third-party libraries from Google can be imported from `@next/third-parties/google`.
+
+### Google Maps Embed
+
+The `GoogleMapsEmbed` component can be used to add a Google Maps Embed to your page. By default, it
+uses the `loading` attribute to lazy-load the embed below the fold.
```
Comment (kara) [inline docs/02-app/01-building-your-application/06-optimizing/10-third-parties.mdx:34]: We should add docs for what each of the attributes mean (e.g. for map mode, we should link to the Google Maps docs here: https://developers.google.com/maps/documentation/embed/embedding-map#choosing_map_modes)

File: docs/02-app/01-building-your-application/06-optimizing/10-third-parties.mdx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  return (
+    <GoogleMapsEmbed
+      apiKey="XYZ"
+      height={200}
+      width="100%"
+      mapMode="place"
+      parameters="q=Brooklyn+Bridge,New+York,NY"
+    />
+  )
+}
```
Comment (kara) [inline docs/02-app/01-building-your-application/06-optimizing/10-third-parties.mdx:97]: This looks like a paste error? Should be `YouTubeEmbed`, no?

```suggestion
import { YouTubeEmbed } from '@next/third-parties/google'

export default function Page() {
  return (
    <YouTubeEmbed
      videoid="ogfYd705cRs"
      height={400}
      width="100%"
      mapMode="place"
      parameters="q=Brooklyn+Bridge,New+York,NY"
    />
  )
}
```

File: docs/02-app/01-building-your-application/06-optimizing/10-third-parties.mdx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+```
+
+</AppOnly>
+
+<PagesOnly>
+
+```jsx filename="pages/index.js"
+import { GoogleMapsEmbed } from '@next/third-parties/google'
```
Comment (kara) [inline docs/02-app/01-building-your-application/06-optimizing/10-third-parties.mdx:105]:
```suggestion
import { YouTubeEmbed } from '@next/third-parties/google'
```

File: docs/02-app/01-building-your-application/06-optimizing/10-third-parties.mdx [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+export default function Page() {
+  return (
+    <YoutubeEmbed
```
Comment (kara) [inline docs/02-app/01-building-your-application/06-optimizing/10-third-parties.mdx:109]: Capitalization is off here

```suggestion
    <YouTubeEmbed
```

File: docs/02-app/01-building-your-application/06-optimizing/10-third-parties.mdx [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    />
+  )
+}
+```
+
+</PagesOnly>
+
+### YouTube Embed
+
+The `YoutubeEmbed` component can be used to load and display a YouTube embed. This component loads
```
Comment (kara) [inline docs/02-app/01-building-your-application/06-optimizing/10-third-parties.mdx:78]:
```suggestion
The `YouTubeEmbed` component can be used to load and display a YouTube embed. This component loads
```

File: docs/02-app/01-building-your-application/06-optimizing/10-third-parties.mdx [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+```jsx filename="pages/index.js"
+import { GoogleMapsEmbed } from '@next/third-parties/google'
+
+export default function Page() {
+  return (
+    <YoutubeEmbed
+      videoid="ogfYd705cRs"
+      height={400}
+      width="100%"
+      mapMode="place"
```
Comment (kara) [inline docs/02-app/01-building-your-application/06-optimizing/10-third-parties.mdx:113]: These attributes look like they're intended for the maps component as well. Does the YT embed have its own attributes we want to demo here?

**Comment by janicklas-ralph** [comment]
I've made the changes on a new PR - https://github.com/vercel/next.js/pull/54805

Please close this PR and review the updated PR

**Pull request closed without merging**